pyccel/parser/semantic.py

```diff
@@ -1052,7 +1054,9 @@ def _visit_DottedName(self, expr, **settings):
                     args  = self._handle_function_args(rhs.args, **settings)
                     func  = first[rhs_name]
                     if new_name != rhs_name:
-                        func  = func.clone(new_name)
+                        if hasattr(func, 'clone'):
+                            func  = func.clone(new_name)
```

Comment:
Did you have a code that failed here?

Reply:
```
def func():
    import numpy as np
    f = np.sqrt(4)
    import math as np
    e = np.sqrt(4)
```

Reply:
Ok. I think this is the right behaviour. Could you add a test. If it works then you can remove the `#TODO`

Reply:
I assume the collision test is for this? But to fully test it I think we also need a collision with a builtin function such as sqrt, not just a user defined function

Reply:
1- Numpy and Math function in pyccel , don't have either clone methods or name property
the same problem is in C and Fortran

Reply:
Yes, but I don't think they need them, so the test that you wrote above should still work shouldn't it?

Reply:
when we remove the condition  it won't work  in C and Fortran 
the reason it was working before is that hidden imports were not taken into consideration  so there was no name clash

Reply:
I didn't suggest removing the condition, just adding a test for the case that doesn't use the condition. Eg the case you gave above

Reply:
only for Fortran and C?

Reply:
Why? Do you expect it to not work for python?
As it is in the semantic stage one language should be sufficient. Python makes the most sense for this PR I think

Reply:
one problem here 
the second import will be Import('math', ['sqrt_001'])
in C and Fortran import are ignored
but in Python we need to print sqrt with the new name sqrt_001
ex  : ```e = sqrt_001(4)```

Reply:
Ah ok I see the problem. That is beyond the scope :+1: